packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/method_channel/method_channel_google_maps_flutter.dart

```diff
@@ -421,8 +421,20 @@ class MethodChannelGoogleMapsFlutter extends GoogleMapsFlutterPlatform {
     CameraUpdate cameraUpdate, {
     required int mapId,
   }) {
-    return channel(mapId).invokeMethod<void>('camera#animate', <String, Object>{
+    return channel(mapId).invokeMethod<void>('camera#animate',
+        <String, dynamic>{'cameraUpdate': cameraUpdate.toJson()});
+  }
+
+  @override
+  Future<void> animateCameraWithConfiguration(
+    CameraUpdate cameraUpdate, {
+    required CameraUpdateAnimationConfiguration configuration,
+    required int mapId,
+  }) {
+    return channel(mapId)
+        .invokeMethod<void>('camera#animate', <String, dynamic>{
       'cameraUpdate': cameraUpdate.toJson(),
+      'duration': configuration.duration?.inMilliseconds,
```

Comment:
We don't generally want to send nulls in collections in method channels.

I would recommend just reverting the changes to this file; we don't use the method channel implementation, it's only still here because removing it requires a breaking change. There's no obligation to implement new features for it.